c10/core/TensorImpl.cpp

```diff
@@ -459,6 +471,12 @@ c10::AutogradMetaInterface* TensorImpl::autograd_meta() const {
 c10::intrusive_ptr<TensorImpl> TensorImpl::shallow_copy_and_detach(
     const c10::VariableVersion& version_counter,
     bool allow_tensor_metadata_change) const {
+  if (key_set_.has(DispatchKey::Python)) {
+    auto r = pyobj_interpreter_.load(std::memory_order_acquire)->detach(this);
```

Comment:
Python key implies there is an interpreter. However, I have to test for interpreter on all inputs in fallback logic, because you can have a mixed function call with Python-available Tensors and Python-unavailable Tensors (remember we OR together the dispatch key sets to determine dispatch).

Reply:
Ho I forgot that case. Sounds good!